### PR TITLE
Migrates to using `@dojo/cli-build-widget` and `@dojo/scripts`

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

Migrates the project to use the Dojo cli command for building both the widget library and custom elements.

The release and testing is provided by `@dojo/scripts`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,14 +55,15 @@
...
     "execa": "^0.9.0",
-    "grunt": "^1.0.1",
-    "grunt-dojo2": "~4.0.0",
-    "grunt-tslint": "^5.0.1",
+    "glob": "7.1.4",
     "husky": "2.2.0",
     "jsdom": "^9.5.0",
     "lint-staged": "8.1.6",
-    "load-grunt-tasks": "^3.5.2",
+    "npm-run-all": "^4.1.5",
```
Comment (mwistrand) [inline package.json]: Is there any reason this shouldn't be pinned?

Comment (agubler) [reply package.json]: missed it! thanks!

File: intern.json [hunk 2]
```diff
@@ -10,16 +10,16 @@
...
+		"./dist/dev/src/common/tests/functional/all.js"
 	],
 
 	"browser": {
 		"loader": {
-			"script": "./node_modules/grunt-dojo2/lib/intern/internLoader.js",
+			"script": "./node_modules/@dojo/scripts/intern/loader.js",
 			"options": {
 				"packages": [
 					{ "name": "src", "location": "_build" }
```
Comment (rorticus) [inline intern.json]: Should this `_build` be `dist/dev/src` as well?

File: src/grid/index.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ import { reference } from '@dojo/framework/core/diff';
 import { Store } from '@dojo/framework/stores/Store';
 import Dimensions from '@dojo/framework/core/meta/Dimensions';
 import Resize from '@dojo/framework/core/meta/Resize';
+import './utils';
```
Comment (rorticus) [inline src/grid/index.ts]: Is this imported for side-effects, or was this not intentional?

File: intern.json (hunk 2 above)
Comment (agubler) [reply intern.json]: probably... I don't know what this does tho lol

File: src/grid/index.ts (hunk 3 above)
Comment (agubler) [reply src/grid/index.ts]: yup this needs to be imported for the module to be included in the library built.

File: intern.json (hunk 2 above)
Comment (mwistrand) [reply intern.json]: Just `dist/dev`, although I don't see any place we're actually using this `browser` config.

**Pull request merged**